examples/00_basic/app.py

```diff
@@ -1,60 +1,121 @@
 import os
 from flask import Flask, jsonify, request, make_response
+from flask_sqlalchemy import SQLAlchemy
 from functools import wraps
+from passlib.hash import sha256_crypt as sc
+import uuid
+import secrets
 import jwt
 import datetime
 
 app = Flask(__name__)
 app.config['SECRET_KEY'] = os.environ.get('Secret')
+app.config['SQLALCHEMY _DATABASE_URI'] = 'something'
 
+db = SQLAlchemy(app)
+blacklist = []
+class User(db.Model):
+    ''' Setting up User'''
+    id = db.Column(db.Integer, primary_key=True)
+    # User vereification
+    email = db.Column(db.String(300), nullable=False, unique=True)
+    username = db.Column(db.String(60), nullable=False, unique=True)
+    password = db.Column(db.String(300), nullable=False)
+    active = db.Column(db.String(50),nullable=False)
+    token = db.Column(db.String(500),nullable = True)
+    # User details
+    name = db.Column(db.String(50), nullable=False)
 
-class User:
-    pass
-    # creating fields for user information
-    # email,#phone number(with county code) can be used as primary keys
-
-
-@app.route('/')
-def home():
-    return 'This is the home page'
-
-
-# creating decorator
-# can be used for email auth
-def token_existence(func):
+#creating decorator
+#can be used for email auth
+def token_existance(func):
     @wraps(func)
-    def checking(*args, **kwargs):
+    def checking(*args,**kwargs):
         token = request.args.get('token')
-        if not token:
-            return jsonify({'message':  'Missing token'}), 403
+        if not token or token in blacklist:
+            return jsonify({'message':  'Missing token'}),403
         try:
             data = jwt.decode(token, app.config['SECRET_KEY'])
+            verified_user = User.query.filter_by(id = data['id']).first()
         except:
-            return jsonify({'message': 'token expired'}), 403
-        return func(*args, **kwargs)
+            return jsonify({'message': 'token is invalid'}),403
+        return func(verified_user,*args, **kwargs)
     return checking
 
+@app.route('/',methods=['GET','POST'])
+def index():
+    return 'Welcome to chatbox'
+
+#go to user home
+@app.route('/home/<token>',methods=['GET','POST'])
+@token_existance
+def user_home(current_user):
+    if current_user.active == "True":
+        return render_template("user_home.html",token=token)
 
+#create/register a user
+@app.route('/register',methods=['GET','POST'])
 def sign_up():
-    pass
-    # create new user
-    # add to database
-    # token can be formed uniquely
+    data = request.form
+
+    #securing the password
+    passw = sc.hash(data['password'])
```

Comment:
Awesome to see the password is being hashed before being stored. In the User class, you can use the `@property` decorator to ensure a property is read-only. Maybe add a `set_password_hash` function to ensure that the string passed is always hashed before being written to the database. This will force new implementers to make sure their passwords are secure.